### PR TITLE
flintlock: init at 0.1.0-alpha.9

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/weaveworks/flintlock/releases/tag/v0.1.0-alpha.9

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by techknowlogick** [comment]
@ofborg eval

File: pkgs/applications/virtualization/flintlock/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+buildGoModule rec{
+  pname = "flintlock";
+  version = "0.1.0-alpha.9";
+
+  src = fetchFromGitHub {
+    owner = "weaveworks";
+    repo = "flintlock";
+    rev = "v${version}";
+    sha256 = "sha256-Xw3g2wh0fPUknSuAKoJL3jxVZS50wSPZ9Wz05zkTVXM=";
```
Comment (milahu) [inline pkgs/applications/virtualization/flintlock/default.nix:19]:
```diff
-    sha256 = "sha256-Xw3g2wh0fPUknSuAKoJL3jxVZS50wSPZ9Wz05zkTVXM=";
+    sha256 = "Xw3g2wh0fPUknSuAKoJL3jxVZS50wSPZ9Wz05zkTVXM=";
```

File: pkgs/applications/virtualization/flintlock/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  vendorSha256 = "sha256-EjVlM6AD+O/z6+R5TRBmmRWbrP4C+qyvsnEjwOkDkUE=";
```
Comment (milahu) [inline pkgs/applications/virtualization/flintlock/default.nix:22]: same

**Review by milahu — APPROVED** [review]

File: pkgs/applications/virtualization/flintlock/default.nix (hunk 1 above)
Comment (techknowlogick) [reply pkgs/applications/virtualization/flintlock/default.nix:19]: wouldn't it be more appropriate to have the hash be the full SRI representation?

Comment (milahu) [reply pkgs/applications/virtualization/flintlock/default.nix:19]: `sha256-` is as useless as `www.`

File: pkgs/applications/virtualization/flintlock/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    makeWrapper
+  ];
+
+  buildInputs = [
+    firecracker
+  ];
+
+  postInstall = ''
+    for prog in flintlockd flintlock-metrics; do
+        wrapProgram "$out/bin/$prog" --prefix PATH : ${lib.makeBinPath [ cni-plugins firecracker containerd runc ]}
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/flintlock/default.nix:38]:
```suggestion
      wrapProgram "$out/bin/$prog" --prefix PATH : ${lib.makeBinPath [ cni-plugins firecracker containerd runc ]}
```

File: pkgs/applications/virtualization/flintlock/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/virtualization/flintlock/default.nix:19]: No, it isn't. `sha256:` can be safely stripped but `sha256-` should be kept.

File: pkgs/applications/virtualization/flintlock/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  src = fetchFromGitHub {
+    owner = "weaveworks";
+    repo = "flintlock";
+    rev = "v${version}";
+    sha256 = "sha256-Xw3g2wh0fPUknSuAKoJL3jxVZS50wSPZ9Wz05zkTVXM=";
+  };
+
+  vendorSha256 = "sha256-EjVlM6AD+O/z6+R5TRBmmRWbrP4C+qyvsnEjwOkDkUE=";
+
+  subPackages = ["cmd/flintlock-metrics" "cmd/flintlockd"];
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/flintlock/default.nix:24]:
```suggestion
  subPackages = [ "cmd/flintlock-metrics" "cmd/flintlockd" ];
```

**Pull request merged**